### PR TITLE
Remove Windows invalid characters from files

### PR DESCRIPTION
**Pull request by StaehliJ**
# Pull request

## Description

The goal of this PR is to allow Microsoft Windows user to check out the project without any issues.

## Changes made

- Remove invalid Windows files name characters

## Checklist

- [x] Your branch has been rebased onto the `main` branch.
- [ ] APIs have been properly documented (if relevant).
- [ ] The documentation has been updated (if relevant).
- [ ] New unit tests have been written (if relevant).
- [ ] The demo has been updated (if relevant).
- [ ] All pull request status checks pass.

### CODE REVIEW

File: pillarbox-core-business/src/androidTest/java/ch/srgssr/pillarbox/core/business/LocalMediaCompositionDataSource.kt [hunk 1]
```diff
@@ -13,10 +13,11 @@ class LocalMediaCompositionDataSource(context: Context) : MediaCompositionDataSo
...
 
     init {
-        val jonsSerializer = Json { ignoreUnknownKeys = true }
-        for (urn in urns) {
-            val json = context.assets.open("$urn.json").bufferedReader().use { it.readText() }
-            localData[urn] = jonsSerializer.decodeFromString(json)
+        val jsonSerializer = Json { ignoreUnknownKeys = true }
+        val json = context.assets.open("media-compositions.json").bufferedReader().use { it.readText() }
+        val listMediaComposition : List<MediaComposition> = jsonSerializer.decodeFromString(json)
+        for(mediaComposition in listMediaComposition){
```
Comment (MGaetan89) [inline pillarbox-core-business/src/androidTest/java/ch/srgssr/pillarbox/core/business/LocalMediaCompositionDataSource.kt:19]:
```suggestion
        val listMediaComposition: List<MediaComposition> = jsonSerializer.decodeFromString(json)
        for (mediaComposition in listMediaComposition) {
```

File: pillarbox-core-business/src/androidTest/java/ch/srgssr/pillarbox/core/business/LocalMediaCompositionDataSource.kt [hunk 2]
```diff
@@ -13,10 +13,11 @@ class LocalMediaCompositionDataSource(context: Context) : MediaCompositionDataSo
     private val localData = HashMap<String, MediaComposition>()
 
     init {
-        val jonsSerializer = Json { ignoreUnknownKeys = true }
-        for (urn in urns) {
```
Comment (MGaetan89) [inline pillarbox-core-business/src/androidTest/java/ch/srgssr/pillarbox/core/business/LocalMediaCompositionDataSource.kt:17]: `urns` is not used anymore, and can be removed.
The other fields in the `companion object` are only used in `CommandersActTrackerTest`, and can probably be moved there?

**Review by MGaetan89 — APPROVED** [review]

**Comment by MGaetan89** [comment]
Should we update our GitHub Actions workflow to build the project on Windows too?

**Comment by StaehliJ** [comment]
> Should we update our GitHub Actions workflow to build the project on Windows too? It could be done in the scope of #339

I'm not sur if there is a window runner that can run virtual android device. This why we have to run androidTest on a macOS runner.

File: pillarbox-core-business/src/androidTest/java/ch/srgssr/pillarbox/core/business/LocalMediaCompositionDataSource.kt (hunk 2 above)
Comment (StaehliJ) [reply pillarbox-core-business/src/androidTest/java/ch/srgssr/pillarbox/core/business/LocalMediaCompositionDataSource.kt:17]: Yes right we can removes `urns` but I will still keep other hardcoded urns for tests here. Has they are linked to the json file that we load in this class.

**Comment by MGaetan89** [comment]
> > Should we update our GitHub Actions workflow to build the project on Windows too? It could be done in the scope of #339
> 
> I'm not sur if there is a window runner that can run virtual android device. This why we have to run androidTest on a macOS runner.

We could run the Android Lint, Detekt and Unit Tests steps in both Ubuntu and Windows, and keep the Android Tests on macOS only.

**Review by MGaetan89 — APPROVED** [review]

**Pull request merged**